src/main/kotlin/calculator/calculator/CalculatorImpl.kt

```diff
@@ -0,0 +1,52 @@
+package calculator.calculator
+
+import calculator.util.isOdd
+
+class CalculatorImpl : Calculator {
+    /**
+     * Node List를 순차적으로 계산한다.
+     */
+    override fun calculate(input: List<Node>): Int {
+        var indexOperator = 1
+        var indexRight = 2
+        var sum = input.getNumber(0)
+
+        while (indexRight != input.size + 1) {
+            val left = sum
+            val operator = input.getOperator(indexOperator)
+            val right = input.getNumber(indexRight)
+            sum = operator.operate(left, right)
+
+            indexRight += 2
+            indexOperator += 2
+        }
+        return sum.value.toInt()
+    }
+
+    /**
+     * Node.Number를 추출한다.
+     * 인덱스가 잘못되었거나 Operator가 들어있다면 NumberIsMissing 을 던진다.
+     */
+    fun List<Node>.getNumber(index: Int): Node.Number {
```

Comment:
`CalculatorImpl` 외에 사용하는 곳이 없어 보이네요. public이 아닌 private으로 바꾸면 어떨까요?

```suggestion
    private fun List<Node>.getNumber(index: Int): Node.Number {
```